packages/website/blog/2024-03-01-parser-options-project-service.md

```diff
@@ -0,0 +1,298 @@
+>   - Most commonly, `.eslintrc.(c)js` files can be tricky to lint, resulting in the dreaded [_TSConfig does not include this file_ error](/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file).
+>
+> We're working on an option to instead call the same TypeScript "Project Service" APIs that editors such as VS Code use to create Programs for us instead.
+> Project Services will automatically detect the TSConfig for each file (like `project: true`), and will also allow type information to be computed for JavaScript files without the `allowJs` compiler option (unlike `project: true`).
+>
+> We hope this option will eventually become the standard way to enable typed linting.
+> However, because it's so new and untested, we're keeping it under the `EXPERIMENTAL_` prefix for all of the `6.??` versions.
+
+Following several months of discussion and testing, we believe the new Project Service API is ready to be used by real-world projects.
+We've found them to be generally faster at runtime and more straightforward to configure.
+
+We're therefore promoting the `parserOptions.EXPERIMENTAL_useProjectService` option to the stable **`parserOptions.projectService`** in typescript-eslint v7.
+
+## Onboarding to the Project Service
+
+You can change over to the new Project Service API by replacing `project` with `projectService` in your ESLint configuration:
+
+<!--tabs-->
+
+### Flat ESLint Config
+
+```diff title="eslint.config.js"
+export default tseslint.flatConfig({
+  // ...
+  languageOptions: {
+    parser: tseslint.parser,
+    parserOptions: {
+-     project: true,
++     projectService: true,
+      tsconfigRootDir: __dirname,
+    },
+  },
+  // ...
+});
+```
+
+### Legacy ESLint Config
+
+```diff title=".eslintrc.cjs"
+module.exports = {
+  // ...
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+-   project: true,
++   projectService: true,
+    tsconfigRootDir: __dirname,
+  },
+  // ...
+}
+```
+
+<!--/tabs-->
+
+Everything else around linting, including running ESLint and configuring rules, should work the same.
+
+### Including Additional Files
+
+One long-standing pain point of typed linting is enabling type information for files not included in the project's `tsconfig.json`, such as an `eslint.config.js` or `vitest.config.ts`.
+Common solutions in the legacy Program API were to either skip type checking for those files or to create a separate `tsconfig.eslint.json` that enabled `compilerOptions.allowJs = true`.
+
+Now, the new Project Service API allows for a configuration object specifying `additionalFiles` globs.
+Those globs allowlist files that should have type information despite not being included in the project's TSConfig.
+
+```js
+parserOptions: {
+  projectService: {
+    additionalFiles: [
+      /* ... */
+    ];
+  }
+}
+```
```

Comment:
nit given there is the full configs below - we can probably skip this block.
Instead just list it as `parserOptions.projectService.additionalFiles`?